guides/keras_nlp/semantic-similarity-with-bert.py

```diff
@@ -0,0 +1,169 @@
+print(f"Total validation samples: {valid_df.shape[0]}")
+print(f"Total test samples: {valid_df.shape[0]}")
+
+"""
+Dataset Overview:
+
+   * sentence1: The premise caption that was supplied to the author of the pair.
+   * sentence2: The hypothesis caption that was written by the author of the pair.
+   * similarity: This is the label chosen by the majority of annotators. Where no majority exists, the label "-" is used (we will skip such samples here).
+
+Here are the "similarity" label values in our dataset:
+
+   * Contradiction: The sentences share no similarity.
+   * Entailment: The sentences have similar meaning.
+   * Neutral: The sentences are neutral.
+
+Let's look at one sample from the dataset:
+
+"""
+
+print(f"Sentence1: {train_df.loc[1, 'sentence1']}")
+print(f"Sentence2: {train_df.loc[1, 'sentence2']}")
+print(f"Similarity: {train_df.loc[1, 'similarity']}")
+
+
+# We have some NaN entries in our train data, we will simply drop them.
+print("Number of missing values")
+print(train_df.isnull().sum())
+train_df.dropna(axis=0, inplace=True)
+
+"""
+### Preprocessing
+"""
+
+print("Train Target Distribution")
+print(train_df.similarity.value_counts())
+
+
+print("Validation Target Distribution")
+print(valid_df.similarity.value_counts())
+
+train_df = (
+    train_df[train_df.similarity != "-"]
+    .sample(frac=1.0, random_state=42)
+    .reset_index(drop=True)
+)
+valid_df = (
+    valid_df[valid_df.similarity != "-"]
+    .sample(frac=1.0, random_state=42)
+    .reset_index(drop=True)
+)
+
+
+train_df["label"] = train_df["similarity"].apply(
+    lambda x: 0 if x == "contradiction" else 1 if x == "entailment" else 2
+)
+y_train = train_df.label
+
+valid_df["label"] = valid_df["similarity"].apply(
+    lambda x: 0 if x == "contradiction" else 1 if x == "entailment" else 2
+)
+y_val = valid_df.label
+
+test_df["label"] = test_df["similarity"].apply(
+    lambda x: 0 if x == "contradiction" else 1 if x == "entailment" else 2
+)
+y_test = test_df.label
+
+preprocessor = BertPreprocessor.from_preset("bert_tiny_en_uncased")
+
+x_train = preprocessor((tf.constant(train_df["sentence1"]), tf.constant(train_df["sentence2"])))
+x_val   = preprocessor((tf.constant(valid_df["sentence1"]), tf.constant(valid_df["sentence2"])))
+x_test = preprocessor((tf.constant(test_df["sentence1"]), tf.constant(test_df["sentence2"])))
+
+"""
+### Train the Model End to End
+ 
+KerasNLP offers pretrained models, we'll use the same here
+"""
+
+model = BertClassifier.from_preset(
+    "bert_tiny_en_uncased", 
+    num_classes=3, 
+    preprocessor=None)
+
+model.compile(loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+              optimizer=tf.keras.optimizers.Adam(1e-5),
+              metrics=["accuracy"])
+
+model.summary()
+
+
+history = model.fit(x = x_train, y = y_train, validation_data = (x_val, y_val), epochs=4, batch_size=64)
+
+
+# In just four epochs we achieved 71% of accuracy.
+
+"""
+### Evaluate model on the test set
```

Comment:
Please add a section showing:

1. saving the model
2. reloading the model
3. performing inference with the model